runtimes/peregrine/src/ismp.rs

```diff
@@ -0,0 +1,103 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at <hello@kilt.io>
+
+use frame_support::parameter_types;
+use ismp::{host::StateMachine, module::IsmpModule, router::IsmpRouter};
+use runtime_common::{AccountId, Balance};
+use sp_core::{ConstU8, Get};
+use sp_std::{boxed::Box, vec::Vec};
+use xcm::v4::Location;
+
+use crate::{
+	governance::{RootOrCollectiveProportion, TechnicalCollective},
+	Balances, Fungibles, Ismp, IsmpParachain, Runtime, RuntimeEvent, Timestamp, TokenGateway, Treasury,
+};
+
+parameter_types! {
+	// The hyperbridge parachain on Polkadot
+	pub const Coprocessor: Option<StateMachine> = Some(StateMachine::Polkadot(3367));
+	pub const HostStateMachine: StateMachine = StateMachine::Polkadot(2086);
+}
+
+#[derive(Default)]
+pub struct Router;
+
+impl IsmpRouter for Router {
+	fn module_for_id(&self, input: Vec<u8>) -> Result<Box<dyn IsmpModule>, anyhow::Error> {
+		match input.as_slice() {
+			pallet_hyperbridge::PALLET_HYPERBRIDGE_ID => Ok(Box::new(pallet_hyperbridge::Pallet::<Runtime>::default())),
+			id if TokenGateway::is_token_gateway(id) => {
+				Ok(Box::new(pallet_token_gateway::Pallet::<Runtime>::default()))
+			}
+			_ => Err(ismp::Error::ModuleNotFound(input))?,
+		}
+	}
+}
+
+impl pallet_ismp::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type AdminOrigin = RootOrCollectiveProportion<TechnicalCollective, 1, 2>;
+	type HostStateMachine = HostStateMachine;
+	type TimestampProvider = Timestamp;
+	type Balance = Balance;
+	type Currency = Balances;
+	type Coprocessor = Coprocessor;
+	type ConsensusClients = (ismp_parachain::ParachainConsensusClient<Runtime, IsmpParachain>,);
+	type Router = Router;
+
+	type WeightProvider = ();
+	type OffchainDB = ();
+}
+
+impl pallet_hyperbridge::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type IsmpHost = Ismp;
+}
+
+impl ismp_parachain::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type IsmpHost = Ismp;
+	type WeightInfo = crate::weights::ismp_parachain::WeightInfo<Runtime>;
+}
+
+pub struct AssetAdmin;
+impl Get<AccountId> for AssetAdmin {
+	fn get() -> AccountId {
+		Treasury::account_id()
+	}
+}
+
+parameter_types! {
+	pub const NativeAssetId: Location = Location::here();
+}
+
+impl pallet_token_gateway::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Dispatcher = Ismp;
```

Comment:
Dispatcher should be pallet-hyperbridge
```suggestion
	type Dispatcher = Ismp;
```

Reply:
[e5a21b4](https://github.com/KILTprotocol/kilt-node/pull/888/commits/e5a21b4f2045c2dbf1c14f3e8bf8823f68ed6f6e)